src/Features/CSharp/Portable/InitializeParameter/InitializeParameterHelpers.cs

```diff
@@ -71,6 +73,14 @@ public static bool IsImplicitConversion(Compilation compilation, ITypeSymbol sou
                     return;
                 }
 
+                if (convertedStatement is ReturnStatementSyntax { Expression: not null } convertedReturn &&
+                    body is ArrowExpressionClauseSyntax arrowClause &&
+                    arrowClause.ArrowToken.TrailingTrivia.IndexOf(SyntaxKind.EndOfLineTrivia) < 0)
+                {
+                    var whiteSpaceTrivia = convertedReturn.Expression.DescendantTrivia().Where(static tr => tr.IsWhitespace() && !tr.IsElastic());
+                    convertedStatement = convertedStatement.ReplaceTrivia(whiteSpaceTrivia, (_, tr) => tr.AsElastic());
```

Comment:
ok.  based on the issues with how some of the tests changed, perhaps we coudl try just putting on elastic trivia on tokens that are at teh start of the line?

that said, i think that still won't necessarily work as my expectation then is that hte formatter would still undo things and place the items on the same line. 

What we genuinely want here is some sort of way of saying "fixup the indentation of these tokens, but otherwise do not touch them".  (or "fixup the indentation of this node, but otherwise don't touch it").

Reply:
actually, since we have the function declaration, we can determinet he desired indentation.  we would then just want to take all leading tokens on every line and ensure it had that indentation if it was just preceded by whitespace on that line.  that seems simpl-ish enough.  importantly, it is targeted, and will likely not have any other downstream impacts.

Reply:
@CyrusNajmabadi I did some checking and first, I am not sure how we know the desired indentation by the function declaration because that's the unconverted version, not the converted one. I don't really see how I can get this information based on that.

But I did checked to see what would happen if I make only the line leading trivia elastic and it behaves unexpectedly, this is the test diff I got on TestUpdateExpressionBody4:

```
Xunit.Sdk.TrueException
Context: Code refactoring application
content of '/0/Test0.cs' did not match. Diff shown with expected as baseline:
 using System;
 
 class C
 {
     int M(int v, string s)
     {
         if (s is null)
         {
             throw new ArgumentNullException(nameof(s));
         }
 
         return v switch
-        {
-            _ => 0
-        };
+{
+    _ => 0
+    };
     }
 }
```
This is very strange. It only reformats correctly if I put every whitespaces elastic.

One thing I am a little perplexed on: the reason the return statement indents correctly is because it is preceded by a 0 width elastic, but if I try to use them here in place, now it removes all indent except the leading underscore one, but only of 4 spaces.

At this point, I am a bit confused if I should look into the formatter's code.

Reply:
My guess is that this is a much larger and more complex item than could be done in a simple bug fix.  Specifically, i think our code-action pipeline needs the concept of "this node needs to be moved, and have its indentation changed, but otherwise have no otehr changes made to it".  We don't have such a concept right now, and it would need to be built out.

Reply:
Hmmm well in that case, I am not sure I can do that myself.

So, it's your call: if this method has too much side effects, feel free to close this. Should someone work on this, they'll have more to go on.